### PR TITLE
feat: add firebase_auth_ui

### PR DESCRIPTION
**Pull request by juanleondev**
## Description

Added Firebase Auth UI to enable different auth functionalities from the package.
- Sign in
- Register
- Forgotten password
- OAuth

https://github.com/elianortega/firebase_app_toolkit/assets/8674244/5938ed7d-ca89-4c96-805e-80012b33c403

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: flutter_news_example/lib/app/router/app_router.dart [hunk 1]
```diff
@@ -33,8 +40,15 @@ class AppRouter {
...
-      pageBuilder: (context, state) => const MaterialPage<void>(
-        child: LoginPage(),
+      builder: (context, state) => SignInScreen(
+        auth: Platform.environment.containsKey('FLUTTER_TEST')
+            ? MockAuth()
+            : null,
+        providers: [
+          EmailAuthProvider(),
+          // TODO(any): add client id to enable desktop login
+          GoogleProvider(clientId: ''),
```
Comment (elianortega) [inline flutter_news_example/lib/app/router/app_router.dart:50]: What is the google provider `clientId` for?

Comment (juanleondev) [reply flutter_news_example/lib/app/router/app_router.dart:50]: This is to enable google sign in desktop, in the docs said that for android and iOS is not necessary

Comment (elianortega) [reply flutter_news_example/lib/app/router/app_router.dart:50]: I think for now we don't need it as the scope of the toolkit is mobile only, I do think later on we can create a desktop/web version but if it is not needed lets remove it

Comment (juanleondev) [reply flutter_news_example/lib/app/router/app_router.dart:50]: the field is required, for that reason I left it with `clientId =''`, if we remove the class we don't have the google sign in button

File: flutter_news_example/lib/app/router/app_router.dart [hunk 2]
```diff
@@ -33,8 +40,15 @@ class AppRouter {
     GoRoute(
       path: LoginPage.path,
       name: LoginPage.path,
-      pageBuilder: (context, state) => const MaterialPage<void>(
-        child: LoginPage(),
+      builder: (context, state) => SignInScreen(
```
Comment (elianortega) [inline flutter_news_example/lib/app/router/app_router.dart:43]: And last question, shouldn't this be wrapped in a `MaterialPage` like the other pages?

Comment (juanleondev) [reply flutter_news_example/lib/app/router/app_router.dart:43]: `builder` doesn't need it.
I think we should avoid pageBuilder unless we want a custom animation in the navigation, MaterialPage is the default with builder and MaterialApp.

ref:

```
/// The widget returned by [builder] is wrapped in [Page] and provided to the
/// root Navigator, the nearest ShellRoute ancestor's Navigator, or the
/// Navigator with a matching [parentNavigatorKey].
///
/// The Page depends on the application type: [MaterialPage] for
/// [MaterialApp], [CupertinoPage] for [CupertinoApp], or
/// [NoTransitionPage] for [WidgetsApp].
```

Comment (elianortega) [reply flutter_news_example/lib/app/router/app_router.dart:43]: Perfect, in that case I'll create another PR removing it 👍

**Review by elianortega — APPROVED** [review]
LGTM

**Pull request merged**